src/main/java/org/folio/innreach/controller/ContributionCriteriaController.java

```diff
@@ -0,0 +1,133 @@
+import org.folio.innreach.domain.entity.ContributionBehavior;
+import org.folio.innreach.domain.service.ContributionCriteriaConfigurationService;
+import org.folio.innreach.dto.ContributionCriteriaDTO;
+import org.folio.innreach.dto.Metadata;
+import org.folio.innreach.mapper.DateMapper;
+import org.folio.innreach.rest.resource.ContributionCriteriaApi;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.HashSet;
+import java.util.UUID;
+
+@Log4j2
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/inn-reach/central-servers/")
+public class ContributionCriteriaController implements ContributionCriteriaApi {
+
+  private final ContributionCriteriaConfigurationService criteriaConfigurationService;
+  private final DateMapper dateMapper;
+
+  @Override
+  @PostMapping("/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> postContributionCriteria(@Valid @RequestBody ContributionCriteriaDTO contributionCriteriaDTO) {
+    return ResponseEntity.status(HttpStatus.CREATED).body(toContributionCriteriaDTO(criteriaConfigurationService.create(toContributionCriteriaConfigurationDTO(contributionCriteriaDTO))));
+  }
+
+
+  @Override
+  @DeleteMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<Void> deleteCriteria(@PathVariable UUID centralServerId) {
+    criteriaConfigurationService.delete(centralServerId);
+    return ResponseEntity.noContent().build();
+  }
+
+  @Override
+  @GetMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> getCriteriaById(@PathVariable UUID centralServerId) {
+    return ResponseEntity.ok(
+      toContributionCriteriaDTO(
+        criteriaConfigurationService.get(centralServerId)
+      )
+    );
+  }
+
+  @Override
+  @PutMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> updateCriteria(@PathVariable UUID centralServerId,
+                                                            @Valid ContributionCriteriaDTO contributionCriteriaDTO) {
+    return ResponseEntity.ok(toContributionCriteriaDTO(
+      criteriaConfigurationService.update(toContributionCriteriaConfigurationDTO(contributionCriteriaDTO))
+    ));
+  }
+
+  public ContributionCriteriaDTO toContributionCriteriaDTO (ContributionCriteriaConfigurationDTO contributionCriteriaConfigurationDTO) {
```

Comment:
1) Why are you doing the DTO conversion if the service is already returning a DTO class?
2) We usually do this transformation not in controllers, but in services.
3) Why do you need a separate method for this? You should use MapStruct converters.